chrome-extension/popup.js

```diff
@@ -0,0 +1,105 @@
+import keys from './config.js';
+
+const NOTES_TEMPLATE_DOC_ID = '1XlcAy-vrleXBxJl5Qy_SxGUyTqcwdUIhyJI2BygpNEc';
+
+const GAPI_CLIENT_URL = 'https://apis.google.com/js/client.js?onload=initGAPI';
+const REVOKE_TOKEN_URL = 'https://accounts.google.com/o/oauth2/revoke?token=';
+
+const DISCOVERY_DOCS = [
+  "https://docs.googleapis.com/$discovery/rest?version=v1",
+  "https://www.googleapis.com/discovery/v1/apis/drive/v3/rest"
+];
+
+let loggedIn = false;
+
+/**
+ * Gets the script for Google APIs client library for browser-side
+ * JavaScript (gapi), sets the button functions, and fills the input. 
+ */
+window.onload = function() {
+  let scriptEl = document.createElement('script');
+  document.head.appendChild(scriptEl);
+  scriptEl.src = GAPI_CLIENT_URL;
+
+  document.getElementById('generate-note-btn').onclick = generateNote;
+  document.getElementById('logout-btn').onclick = logout;
+  document.getElementById('doc-name-input').value = 'gNote ' + getDate();
+}
+
+/** Initializes gapi. */
+window.initGAPI = function initGAPI() {
+  gapi.client.init({
+    apiKey: keys.API_KEY,
+    discoveryDocs: DISCOVERY_DOCS,
+  }).then(handleLogin)
+  .catch(error => console.log('Error:', error));
+}
+
+/**
+ * Requests user login (if necessary) then sets gapi access token
+ * and displays user info in popup.
+ */
+function handleLogin() {
+  chrome.identity.getAuthToken({interactive: true}, function(token) {
+    if (chrome.runtime.lastError) {
+      return;
+    }
+    gapi.auth.setToken({
```

Comment:
Quick question: where are the `chrome` and `gapi` variables coming from? :0

Reply:
I think the `chrome` variable is built into the chrome extension (not entirely sure about this). But in order to use `chrome.identity`, you would need to add `identity` to the `permissions` in the manifest. And I believe the `gapi` variable comes from the script that is loaded from here: https://apis.google.com/js/client.js?onload=initGAPI (`GAPI_CLIENT_URL`). The onload part is added since we can't use `gapi` until the script is fully loaded.